e2e/testcafe-devextreme/tests/dataGrid/common/virtualColumns.ts

```diff
@@ -296,6 +296,67 @@ test('Columns should be rendered correctly after reinit of columns controller',
   },
 }));
 
+test('Columns reordering should work with virtual columns', async (t) => {
+  const dataGrid = new DataGrid('#container');
+  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+
+  await dataGrid.scrollTo(t, { x: 100000 });
+
+  await t.drag(
+    dataGrid.getHeaders().getHeaderRow(0).getHeaderCell(499).element,
+    -110,
+    0,
+  );
+
+  await t.debug();
+
+  await t
+    .expect(await takeScreenshot('data-grid__virtual-columns__reoder.png', dataGrid.element))
+    .ok()
+    .expect(compareResults.isValid())
+    .ok(compareResults.errorMessages());
+}).before(async () => createWidget('dxDataGrid', {
+  allowColumnReordering: true,
+  height: 440,
+  dataSource: generateData(150, 500),
+  columns: generateColumns(500),
+  columnWidth: 100,
+  scrolling: {
+    columnRenderingMode: 'virtual',
+  },
+}));
+
+test('Grouping should work with virtual columns', async (t) => {
+  const dataGrid = new DataGrid('#container');
+  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+
```

Comment:
I recommend adding a dataGrid.isReady check for greater test stability.

Reply:
Let's postpone it for general. If I add it here now – we will need to find it and delete in future